### ISSUE TITLE
Feature request: A way to profile the performance of XAML bindings

### ISSUE THREAD

**Issue by rimex-bkimmett** [dotnet/maui#16028]
### Description

I would want a profiler specifically for testing the performance of XAML bindings in .NET MAUI. The current profiler simply breaks things down by function without regard to XAML; this new profiler would list XAML binding(s) under investigation, how long each one took from start to completion, and what function calls/lines of code in a specific binding took the most time.

Ideally, the new profiler would be freely available (the Xamarin profiler did something like this, but I only ever had Visual Studio Professional, not Enterprise).

### Public API Changes

There would be no API changes, as this would be a change to available tooling.

### Intended Use-Case

I have been working on optimizing an app that was originally written in Xamarin, and is now migrated to .NET MAUI. One page of the app has numerous XAML bindings - it's also very slow. I suspect some of the bindings in particular are slowing the page down - but I don't know which. A tool that would let me find slow bindings, and what part of a binding is slow, would be a great help.

**Comment by marcojak** [comment]
This seems very interesting. Anything that could lead to performance improvement should always prioritised as one of the biggest issue with Xamarin e Maui is their slowness.

**Comment by mattleibow** [comment]
@jonathanpeppers thoughts on this and things we can do to add better metrics?

**Comment by jonathanpeppers** [comment]
Can we attach a `.speedscope` or `.nettrace` file here as an example? https://aka.ms/profile-maui

If there isn't sufficient data there, this might be something we'd better off filing at https://github.com/dotnet/diagnostics for the .NET diagnostics team to think about.

**Comment by rimex-bkimmett** [comment]
I see the needs-info label, but what's the question? 

For reference, I've done profiling on .NET MAUI on Android before, but not only does it not show up in VS's live profiling tools directly (and was a pain to set up), I wasn't able to associate performance costs with specific bindings in the profiler results - the profiler is not aware of bindings as a concept, generally.

**Comment by kubaflo** [comment]
You are right: no specific unanswered question accompanied `s/needs-info`, so that label was not appropriate and has been removed. The proposal remains open for product evaluation; your profiling context is useful evidence.

> Disclosure: this comment was generated by AI.

**Comment by kubaflo** [comment]
@rimex-bkimmett — following up with something more useful than my previous reply, which answered the label question and ignored your actual point.

Your observation is really the core of this request: the existing tooling isn't binding-aware. A `.speedscope` / `.nettrace` capture — which @jonathanpeppers asked about back in 2023 — will show time spent inside the binding infrastructure, but it can't attribute that cost to *a specific binding expression*. That gap is precisely what makes XAML binding performance hard to diagnose in practice, and it's why "just profile it" isn't a substitute for this feature.

I've recorded that as the justification for the proposal. It stays open as `proposal/open`, and nothing further is needed from you.

> Disclosure: this comment was generated by AI.
